torch/_inductor/lowering.py

```diff
@@ -3530,29 +3530,25 @@ def pow_native(a, b):
     return ops.pow(a, b)
 
 
-def _is_ir_node_and_cuda(x):
-    if isinstance(x, ir.IRNode) and decode_device(x.get_device()).type == "cuda":
-        return True
-
-    return False
+fallback_pow = fallback_handler(aten.pow)
 
 
 @register_lowering(aten.pow, broadcast=True)
 def pow(a, b):
-    if _is_ir_node_and_cuda(a) and _is_ir_node_and_cuda(b):
-        assert a.get_dtype() in (
-            torch.float16,
-            torch.float32,
-            torch.float64,
-        ), "Pow input must be floating point."
     if isinstance(b, float) and b == int(b):
         return pow(a, int(b))
     elif isinstance(b, float) and b == 0.5:
         return sqrt(a)
     elif isinstance(b, int) and b == 1:
         return a
-    elif isinstance(b, int) and -32 < b < 32:
-        # Optimize away small fixed powers
+
+    dtype = a.get_dtype() if isinstance(a, ir.TensorBox) else b.get_dtype()
+    is_integer_pow = is_integer_dtype(dtype)
+    # Optimize away small fixed powers, or for integers inline the simple cases
+    embed_exponent = isinstance(b, int) and (
+        -32 < b < 32 or (is_integer_pow and b >= 0)
```

Comment:
What's the rationale behind `is_integer_pow and b >= 0`? Since `a` and `b` cannot both be scalars, I guess that `dtype` here is coming from `a`. If this is the case, is it alright that we would want to inline the computation of `a ** 12309481`? And if so, why don't we want to inline that of `a ** -12309481` as `1. / (a ** 12309481)`?